lib/cc/cli.rb

```diff
@@ -17,15 +17,25 @@ module CLI
     autoload :ValidateConfig, "cc/cli/validate_config"
     autoload :Version, "cc/cli/version"
 
-    def self.debug(message, values = {})
-      if ENV["CODECLIMATE_DEBUG"]
-        if values.any?
-          message << " "
-          message << values.map { |k, v| "#{k}=#{v.inspect}" }.join(" ")
+    class Logger
+      def debug(message, values = {})
+        if ENV["CODECLIMATE_DEBUG"]
+          if values.any?
+            message << " "
+            message << values.map { |k, v| "#{k}=#{v.inspect}" }.join(" ")
+          end
+
+          # Leading and trailing newlines to prevent mangling
+          $stderr.print("\n[DEBUG] #{message}\n")
         end
+      end
 
-        $stderr.puts("[DEBUG] #{message}")
+      # Any non-DEBUG output is handled not via logging
+      def method_missing(*)
```

Comment:
Is this actually used right now?  Since `#debug` doesn't follow the same contract as ruby's `Logger`, I don't see a reason for a shim like this unless we already have usage relying on it (which I don't believe we do).